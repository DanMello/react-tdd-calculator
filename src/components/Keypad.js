import React, { Component } from 'react';
import PropTypes from 'prop-types';
// import styles from 'styles/Display.css';

class Keypad extends Component {

  render() {

    return (
      <div>Hello</div>
    );
  };
};

Keypad.propTypes = {
  callOperator: PropTypes.func.isRequired,
  numbers: PropTypes.array.isRequired,
  operators: PropTypes.array.isRequired,
  setOperator: PropTypes.func.isRequired,
  updateDisplay: PropTypes.func.isRequired,
}

export default Keypad;