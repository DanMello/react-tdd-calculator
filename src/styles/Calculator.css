.calculatorContainer {
  background: #3a3d50;
  min-height: 400px;
  border-radius: 5px;
  padding-left: 5px;
  padding-right: 5px;
  padding-top: 5px;
  margin: 10px;
};