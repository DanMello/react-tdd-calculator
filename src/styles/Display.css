.displayContainer {
  padding-left: 5px;
  padding-right: 5px;
  padding-top: 30px;
  padding-bottom: 30px;
  background: white;
  display: flex;
  justify-content: flex-end;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
  font-size: 60px;
  font-family: sans-serif;
};