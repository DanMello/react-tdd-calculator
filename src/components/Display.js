import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from 'styles/Display.css';

class Display extends Component {

  render() {

    return (
      <div className={styles.displayContainer}>
        {this.props.displayValue}
      </div>
    );
  };
};

Display.propTypes = {
  displayValue: PropTypes.string.isRequired
};

export default Display;