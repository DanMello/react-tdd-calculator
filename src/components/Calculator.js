import React, { Component } from 'react';
import Display from './Display';
import Keypad from './Keypad';
import styles from 'styles/Calculator.css';

export default class Calculator extends Component {

  constructor() {
    super()

    this.state = {
      displayValue: '0',
      numbers: [],
      operators: [],
      selectedOperator: '',
      storedValue: ''
    };

    this.callOperator = this.callOperator.bind(this)
    this.setOperator = this.setOperator.bind(this)
    this.updateDisplay = this.updateDisplay.bind(this)
  };

  callOperator() {
    console.log('call operator')
  };

  setOperator() {
    console.log('set operation')
  };

  updateDisplay() {
    console.log('update display')
  };

  render() {
    return (
      <div className={styles.calculatorContainer}>
        <Display displayValue={this.state.displayValue} />
        <Keypad
          callOperator={this.callOperator}
          numbers={this.state.numbers}
          operators={this.state.operators}
          setOperator={this.setOperator}
          updateDisplay={this.updateDisplay}
        />
      </div>
    );
  };
};