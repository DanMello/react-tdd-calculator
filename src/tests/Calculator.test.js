import React from 'react';
import ReactDOM from 'react-dom';
import Calculator from '../components/Calculator';
import Display from '../components/Display';
import Keypad from '../components/Keypad';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

it('Renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Calculator />, div);
});

it('Renders correctly', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Should render a <div />', () => {

  const wrapper = shallow(<Calculator />);

  expect(wrapper.find('div').length).toEqual(1);
});

it('should render the Display and Keypad Components with their props', () => {
  
  const wrapper = shallow(<Calculator />);

  expect(wrapper.containsAllMatchingElements([
    <Display displayValue={wrapper.instance().state.displayValue}/>,
    <Keypad 
      callOperator={wrapper.instance().callOperator}
      numbers={wrapper.instance().state.numbers}
      operators={wrapper.instance().state.operators}
      setOperator={wrapper.instance().setOperator}
      updateDisplay={wrapper.instance().updateDisplay}
    />
  ])).toEqual(true);
});