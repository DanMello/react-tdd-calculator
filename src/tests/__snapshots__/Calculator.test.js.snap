// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="calculatorContainer"
>
  <div
    className="displayContainer"
  >
    0
  </div>
  <div>
    Hello
  </div>
</div>
`;
